### PR TITLE
Simplified link tracking related tables naming

### PR DESCRIPTION
**Pull request by allouis**
- Removes superfluous "link" from table names
- Fixes type definititon of dropTables util
- Updates & renames models
- Noop existing migrations to avoid unnecessary work

### CODE REVIEW

File: ghost/core/core/server/data/migrations/utils/tables.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ function addTable(name, tableSpec) {
 /**
  * Creates migration which will drop a table
  *
- * @param {[string]} names  - names of the tables to drop
```
Comment (allouis) [inline ghost/core/core/server/data/migrations/utils/tables.js:41]: `[string]` means that we accept an array with exactly one string in, whereas `string[]` is an array of zero or more strings

**Comment by SimonBackx** [comment]
What do you think about the `LinkRedirectRepository` and `LinkRedirect` names?

File: ghost/core/core/server/data/migrations/versions/5.16/2022-09-19-09-04-add-link-redirects-table.js [hunk 2]
```diff
@@ -1,10 +1,4 @@
...
-    id: {type: 'string', maxlength: 24, nullable: false, primary: true},
-    from: {type: 'string', maxlength: 2000, nullable: false},
-    to: {type: 'string', maxlength: 2000, nullable: false},
-    post_id: {type: 'string', maxlength: 24, nullable: true, unique: false, references: 'posts.id', setNullDelete: true},
-    created_at: {type: 'dateTime', nullable: false},
-    updated_at: {type: 'dateTime', nullable: true}
-});
+module.exports = {
+    async up() {},
+    async down() {}
```
Comment (SimonBackx) [inline ghost/core/core/server/data/migrations/versions/5.16/2022-09-19-09-04-add-link-redirects-table.js:3]: Maybe we need log messages here?

Comment (allouis) [reply ghost/core/core/server/data/migrations/versions/5.16/2022-09-19-09-04-add-link-redirects-table.js:3]: Good shout 👍

Comment (allouis) [reply ghost/core/core/server/data/migrations/versions/5.16/2022-09-19-09-04-add-link-redirects-table.js:3]: Channelling your inner @daniellockyer 😝

Comment (SimonBackx) [reply ghost/core/core/server/data/migrations/versions/5.16/2022-09-19-09-04-add-link-redirects-table.js:3]: Learning from experience 😅

Comment (daniellockyer) [reply ghost/core/core/server/data/migrations/versions/5.16/2022-09-19-09-04-add-link-redirects-table.js:3]: 😆

File: ghost/core/core/server/data/migrations/versions/5.17/2022-09-27-13-53-remove-click-tracking-tables.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const {dropTables} = require('../../utils');
+
+module.exports = dropTables(['link_redirects', 'members_link_click_events']);
```
Comment (daniellockyer) [inline ghost/core/core/server/data/migrations/versions/5.17/2022-09-27-13-53-remove-click-tracking-tables.js:3]: This produces an irreversible migration, which is only something we do in majors. We'll have to switch to some other utils, or only use the `.up` from this

**Review by ErisDS — APPROVED** [review]
I have no additional notes.

File: ghost/core/core/server/data/migrations/versions/5.17/2022-09-27-13-53-remove-click-tracking-tables.js (hunk 3 above)
Comment (allouis) [reply ghost/core/core/server/data/migrations/versions/5.17/2022-09-27-13-53-remove-click-tracking-tables.js:3]: Aha okay!

**Review by daniellockyer — APPROVED** [review]

**Pull request merged**